esmvalcore/preprocessor/_volume.py

```diff
@@ -275,6 +277,51 @@ def volume_statistics(cube, operator):
     return _create_cube_time(src_cube, result, times)
 
 
+def axis_statistics(cube, axis, operator):
+    """Perform statistics along a given axis.
+
+    Operates over an axis direction. If weights are required,
+    they are computed using the coordinate bounds.
+
+    Arguments
+    ---------
+    cube: iris.cube.Cube
+        input cube.
+    axis: str
+        direction over where to apply the operator.
+    operator: str
+        statistics to perform.
+
+    Returns
+    -------
+    iris.cube.Cube
+        collapsed cube.
+    """
+    try:
+        coord = cube.coord(axis=axis)
+    except iris.exceptions.CoordinateNotFoundError as err:
+        raise ValueError(
+            'Axis {} not found in cube {}'.format(
+                axis,
+                cube.summary(shorten=True))) from err
+    operation = get_iris_analysis_operation(operator)
+    weights = None
+    if operator_accept_weights(operator):
+        coord_dim = cube.coord_dims(cube.coord(axis=axis))[0]
```

Comment:
```suggestion
        coord_dims = cube.coord_dims(coord)
        assert len(coord_dims) == 1
        coord_dim = coord_dims[0]
```
Note that also the weights calculation below is not meaningful if the coordinate is multi-dimensional.

Reply:
> if the coordinate is multi-dimensional

What should the behaviour be if the coordinate is multidimensional? Because for instance in the case of wanting to collapse the 'X' axis for a multidimensional coordinate, this implementation would collapse the 'Y' axis instead because the value of `coord_dim` would always be 1

Reply:
Actually, in that case `len(coord_dims)` would be 2 and thus this would raise an `AssertionError`. This is the crudest way to say we only support 1d coordinates right now. Since we are shooting for a quick solution, I think supporting only 1d coordinates is sensible, but it would be better to be clearer about it, for example by raising a `NotImplemented` exception.

I would probably go so far as to do that even in the case of weight free operators, at least to start with.

Reply:
Ok, added the raise of the NotImplementedError